pxr/usdImaging/usdImaging/primAdapter.cpp

```diff
@@ -768,12 +768,16 @@ UsdImagingPrimAdapter::_MergePrimvar(
     bool indexed) const
 {
     HdPrimvarDescriptor primvar(name, interp, role, indexed);
-    HdPrimvarDescriptorVector::iterator it =
-        std::find(vec->begin(), vec->end(), primvar);
-    if (it == vec->end())
-        vec->push_back(primvar);
-    else
-        *it = primvar;
+
+    for (HdPrimvarDescriptorVector::iterator it = vec->begin();
```

Comment:
If you're open to the suggestion, I think you can preserve the existing structure and the usage of the STL with `std::find_if`.

Either way, it's great to see this fix!